MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/extrusions/RotationExtrusionActivity.java

```diff
@@ -0,0 +1,230 @@
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.fillExtrusionColor;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.fillExtrusionHeight;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.fillExtrusionOpacity;
+
+public class RotationExtrusionActivity extends AppCompatActivity implements SensorEventListener{
+
+    private MapView mapView;
+    private MapboxMap mapboxMap;
+    private Light light;
+    private boolean isMapAnchorLight;
+    private boolean isLowIntensityLight;
+    private boolean isRedColor;
+    private boolean isInitPosition;
+    private LocationEngine locationEngine;
+    private LocationLayerPlugin locationLayerPlugin;
+    private SensorManager sensorManager;
+    private Sensor gyro;
+    private Sensor magnetic;
+    float[] mGravity;
+    float[] mGeomagnetic;
+    float azimut;
+    float pitch;
+    float roll;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Mapbox access token is configured here. This needs to be called either in your application
+        // object or in the same activity which contains the mapview.
+        Mapbox.getInstance(this, getString(R.string.access_token));
+
+        setContentView(R.layout.activity_extrusion_rotation);
+
+        mapView = (MapView) findViewById(R.id.mapView);
+        mapView.onCreate(savedInstanceState);
+        mapView.getMapAsync(new OnMapReadyCallback() {
+            @Override
+            public void onMapReady(@NonNull final MapboxMap map) {
+                mapboxMap = map;
+                setupBuildings();
+                setupLocationPlugin();
+            }
+        });
+
+        //initialize sensors
+        sensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
+
+        gyro = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+        magnetic = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD);
+
+        sensorManager.registerListener(this, gyro , 200);
+        sensorManager.registerListener(this, magnetic , 200);
+    }
+
+    private void setupBuildings() {
+        FillExtrusionLayer fillExtrusionLayer = new FillExtrusionLayer("3d-buildings", "composite");
+        fillExtrusionLayer.setSourceLayer("building");
+        fillExtrusionLayer.setFilter(eq("extrude", "true"));
+        fillExtrusionLayer.setMinZoom(15);
+        fillExtrusionLayer.setProperties(
+                fillExtrusionColor(Color.LTGRAY),
+                fillExtrusionHeight(Function.property("height", new IdentityStops<Float>())),
+                fillExtrusionBase(Function.property("min_height", new IdentityStops<Float>())),
+                fillExtrusionOpacity(0.9f)
+        );
+        mapboxMap.addLayer(fillExtrusionLayer);
+    }
+
+    private void setupLocationPlugin(){
+//        locationEngine = new LocationSource(this);
```

Comment:
Do you want to include the location icon in the example or no? If not, then please delete the commented out code.